pylearn2/datasets/dataset.py

```diff
@@ -18,7 +18,7 @@ def __iter__(self):
         return self.iterator()
 
```

Comment:
Done.
I had to update the signature of `get_batch_topo` to get consistant with the implementation in `DenseDesignMatrix`, and took most of the documentation from `get_batch_design`.
Ultimately, I think these methods should be deprecated and disappear, maybe being replaced by one that requests a list of sources and spaces.